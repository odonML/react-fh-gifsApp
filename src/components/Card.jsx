import PropTypes from 'prop-types'
function Card({ url, title }) {
	return (
		<div className="card">
			<img src={url} alt={title} />
			<p>{title}</p>
		</div>
	);
}
Card.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
}

export default Card;
