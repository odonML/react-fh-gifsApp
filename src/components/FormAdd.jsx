import PropTypes from 'prop-types'
import { useState } from 'react';

function FormAdd({onAddCategori}) {
    const [inpitValue, setInpitValue] = useState("");

    const onInputChange = ({target}) => {
        setInpitValue(target.value)
    }
    const addCategori = (event) =>{
        event.preventDefault();
        if(inpitValue.trim().length < 1) return;
        onAddCategori(inpitValue)
        setInpitValue("")
    }
	return (
		<form onSubmit={addCategori} aria-label='form'>
			<input type="text" placeholder="Buscar Gifs" value={inpitValue} onChange={ onInputChange}/>
		</form>
	);
}

FormAdd.propTypoes = {
    onAddCategori: PropTypes.func.isRequired,
}


export default FormAdd;