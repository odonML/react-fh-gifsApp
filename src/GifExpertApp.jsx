import { useState } from "react";
import {List, FormAdd} from "./components";


function GifExpertApp() {
	const [categories, setCategories] = useState(["one pounch", "dragon ball"]);

	const onAddCategori = (newCategori) => {
        if(categories.includes(newCategori)) return; 
		setCategories([...categories, newCategori]);
	};

	return (
		<>
			<h1>Hola mundo</h1>
			<FormAdd onAddCategori={onAddCategori} />
			<List categories={categories}></List>
		</>
	);
}

export default GifExpertApp;
