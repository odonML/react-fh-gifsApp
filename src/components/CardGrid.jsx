import { useFetchGifs } from "../hooks/useFetchGifs";
import Card from "./Card";
import PropTypes from 'prop-types'


function CardGrid({ categori }) {
	const {images, isLoading} = useFetchGifs(categori);	

	return (
		<>
			<h3>{categori}</h3>
			{
				isLoading && <h2>Cargando...</h2>
			}
			<div className="card-grid">
				{images.map((img) => (
					<Card key={img.id} {...img}></Card>
				))}
			</div>
		</>
	);
}

CardGrid.propTypes = {
	categori: PropTypes.string.isRequired,
}

export default CardGrid;
