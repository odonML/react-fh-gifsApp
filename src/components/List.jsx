import PropTypes from 'prop-types'
import CardGrid from './CardGrid';
function List({categories}) {
    return (
        <>
        {
            categories.map((categori) => (
                <CardGrid key={categori} categori={categori}></CardGrid>
            ))
        }
        </>
      );
}

List.propTypoes = {
    categories: PropTypes.array.isRequired,
}

export default List;