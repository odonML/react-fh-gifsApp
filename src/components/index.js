import Card from "./Card";
import CardGrid from "./CardGrid";
import FormAdd from "./FormAdd";
import List from "./List";

export { Card, CardGrid, FormAdd, List };
