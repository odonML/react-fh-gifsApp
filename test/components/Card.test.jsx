import { render, screen } from "@testing-library/react";
import { Card } from "../../src/components";

describe('unit test of Card component', () => {
    const title = "pitbulls baby"
    const url = "http://www.ababa.com/"
    test('should match snapshot', () => { 
        const {container} = render(<Card title={title} url={url}></Card>)
        expect(container).toMatchSnapshot();
     })

     test("show url and title", () => {
         render(<Card title={title} url={url}></Card>);
         const {src, alt} = screen.getByRole("img");
        expect(src).toEqual(url)
        expect(alt).toEqual(title)
     })

     test('should show title in p', () => { 
         render(<Card title={title} url={url}></Card>);
        
        expect(screen.getByText(title)).toBeTruthy()
      })
});