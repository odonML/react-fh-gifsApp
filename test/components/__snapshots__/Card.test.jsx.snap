// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit test of Card component should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="pitbulls baby"
      src="http://www.ababa.com/"
    />
    <p>
      pitbulls baby
    </p>
  </div>
</div>
`;
